framework/src/meshgenerators/FillBetweenSidesetsGenerator.C

```diff
@@ -0,0 +1,239 @@
+  auto input_mesh_2 = dynamic_cast<ReplicatedMesh *>(_input_2.get());
+  if (!input_mesh_1)
+    paramError("input_mesh_1", "Input is not a replicated mesh, which is required.");
+  if (*(input_mesh_1->elem_dimensions().begin()) != 2 ||
+      *(input_mesh_1->elem_dimensions().rbegin()) != 2)
+    paramError("input_mesh_1", "Only 2D meshes are supported.");
+  if (!input_mesh_2)
+    paramError("input_mesh_2", "Input is not a replicated mesh, which is required.");
+  if (*(input_mesh_2->elem_dimensions().begin()) != 2 ||
+      *(input_mesh_2->elem_dimensions().rbegin()) != 2)
+    paramError("input_mesh_2", "Only 2D meshes are supported.");
+
+  MeshTools::Modification::translate(
+      *input_mesh_1, _mesh_1_shift(0), _mesh_1_shift(1), _mesh_1_shift(2));
+  MeshTools::Modification::translate(
+      *input_mesh_2, _mesh_2_shift(0), _mesh_2_shift(1), _mesh_2_shift(2));
+
+  const auto input_mesh_1_external_bids =
+      MooseMeshUtils::getBoundaryIDs(*input_mesh_1, _boundary_1, false);
+  const auto input_mesh_2_external_bids =
+      MooseMeshUtils::getBoundaryIDs(*input_mesh_2, _boundary_2, false);
+
+  for (unsigned int i = 1; i < input_mesh_1_external_bids.size(); i++)
+  {
+    MooseMeshUtils::changeBoundaryId(
+        *input_mesh_1, input_mesh_1_external_bids[i], input_mesh_1_external_bids.front(), true);
+  }
+  for (unsigned int i = 1; i < input_mesh_2_external_bids.size(); i++)
+  {
+    MooseMeshUtils::changeBoundaryId(
+        *input_mesh_2, input_mesh_2_external_bids[i], input_mesh_2_external_bids.front(), true);
+  }
+
+  if (!FillBetweenPointVectorsTools::isExternalBoundary(*input_mesh_1,
+                                                        input_mesh_1_external_bids.front()))
+    paramError("boundary_1", "The boundary provided is not an external boundary.");
+  if (!FillBetweenPointVectorsTools::isExternalBoundary(*input_mesh_2,
+                                                        input_mesh_2_external_bids.front()))
+    paramError("boundary_2", "The boundary provided is not an external boundary.");
+
+  Real max_input_mesh_1_node_radius;
+  Real max_input_mesh_2_node_radius;
+  std::vector<dof_id_type> boundary_1_ordered_nodes;
+  std::vector<dof_id_type> boundary_2_ordered_nodes;
+
+  try
+  {
+    FillBetweenPointVectorsTools::isBoundaryValid(*input_mesh_1,
+                                                  max_input_mesh_1_node_radius,
+                                                  boundary_1_ordered_nodes,
+                                                  _mesh_1_shift,
+                                                  input_mesh_1_external_bids.front());
+  }
+  catch (MooseException & e)
+  {
+    if (((std::string)e.what())
+            .compare("This mesh generator does not work for the provided external boundary as it "
+                     "is not a closed loop.") != 0)
+      paramError("boundary_1", "The provided boundary is not a single-segment boundary.");
```

Comment:
so if another exception is thrown what happens?

Reply:
The boundary is usable to `FillBetweenSidesetsGenerator` only if this exception (invalid_type #2 in previous version) is thrown by `isBoundaryValid()`. The "valid" in `isBoundaryValid()` is for `PeripheralRingMeshGenerator`.

It occurred to me that another error message is needed when isBoundaryValid()'s return is `true`. Added.

(Maybe I need to change the method name to `isBoundaryClosedLoop()`?

Reply:
Renamed as `isBoundarySimpleClosedLoop()` in the latest commit. Now I think it makes more sense here.

Reply:
umm good point.
My concern was that if the routine throws with a different exception, then the if is not triggered and we just move forward with no error

Reply:
I think that is still the case?

Reply:
In the latest commit (see Lines 154 thru 170), there are following four possible cases here:

(1) isBoundarySimpleClosedLoop() returns `true`:  This means that the boundary is a closed loop, which is not supported by `FillBetweenSideSetsGenerator`. Then `paramError()`  in the `try` bracket is triggered and throw **"The provided boundary is closed loop, which is not supported."**

If isBoundarySimpleClosedLoop() returns `false`, one of the three exception messages is thrown:
(2) "This mesh generator does not work for the provided external boundary as it has more than one segments." The `paramError()` in the `catch` bracket is triggered and throw **"The provided boundary is not an open single-segment boundary."** 

(3) "This mesh generator does not work for the provided external boundary as it is not a closed loop." This is the only case in which `FillBetweenSideSetsGenerator` works (i.e., an **open and one-segment boundary**). It will pass the `catch` bracket without triggering `paramError()`

(4) "This mesh generator does not work for the provided external boundary as azimuthal angles of consecutive nodes do not change monotonically." The `paramError()` in the `catch` bracket is triggered and throw **"The provided boundary is not an open single-segment boundary."** 

I think this covers all the scenarios.

Reply:
ok right now it covers all the scenarios. But these tools will change. I think you should have a generic catcher for when an exception is thrown that has not been hardcoded in the downstream object. 
It's an else to the if that you added

Reply:
Here the only `else` is that the exception message "This mesh generator does not work for the provided external boundary as it is not a closed loop." is thrown. In that case, the `else` is what `FillBetweenSideSetsGenerator` wants.

But I do understand your concern. I think I will move the `try` and `catch` here into a new function called `isOpenSingleSegmentBoundary()` in the tools namespace to eliminate confusion here.

Reply:
but then you will need to pass name/type to the toolbox to make a clear error message.

I think the 
```
try
{
}
catch
{
  if (this known exception)
    this_message
  else
    return this other exception message that wasnt planned when the class was created
}
```

is fine

Reply:
I used another set of moose exceptions to prevent name/type passing. See the commit and let me know your comment. Thanks.